### PR TITLE
Update docs for CoreDNS newer than 1.3.2

### PR DESCRIPTION
**Pull request by johscheuer**
Please provide a description for what this PR is for.

This PR provides documentation for CoreDNS users that use a more recent version of CoreDNS (> 1.3.2). Since the `proxy` command was deprecated the resulting `ConfigMap` for `CoreDNS` looks a little bit different.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by johscheuer** [comment]
ping :)

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 1]
```diff
@@ -160,7 +160,7 @@ data:
 EOF
 {{< /text >}}
 
-For clusters that use CoreDNS:
+For clusters that use CoreDNS (< 1.4.0):
```
Comment (adammil2000) [inline content/en/docs/setup/install/multicluster/gateways/index.md]: Please use tabs for each option (kube-dns, CoreDNS 1.4.0 and earlier, and CoreDNS 1.4.0 and later. The way to use tabs is described here: https://istio.io/about/contribute/creating-and-editing-pages/#using-tabs

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 2]
```diff
@@ -194,6 +194,40 @@ data:
 EOF
 {{< /text >}}
 
+For clusters that use CoreDNS (> 1.4.0):
```
Comment (adammil2000) [inline content/en/docs/setup/install/multicluster/gateways/index.md]: Please implement this in-line YAML code using the snippets mechanism so we can implement automated samples testing. The way to do that is described here: https://istio.io/about/contribute/creating-and-editing-pages/#snippets

Thanks!

File: content/en/docs/setup/install/multicluster/gateways/index.md (hunk 1 above)
Comment (johscheuer) [reply content/en/docs/setup/install/multicluster/gateways/index.md]: Oh I fixed it, I actually meant `>= 1.4.0`.

File: content/en/docs/setup/install/multicluster/gateways/index.md (hunk 2 above)
Comment (johscheuer) [reply content/en/docs/setup/install/multicluster/gateways/index.md]: Where do I place the snippet file (I couldn't find any reference) ?

Comment (johscheuer) [reply content/en/docs/setup/install/multicluster/gateways/index.md]: ping @adammil2000 could you tell me where I should put the files ?

**Comment by geeknoid** [comment]
@frankbu Does this look good to you? Should we merge this?

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 3]
```diff
@@ -145,8 +145,8 @@ Create one of the following ConfigMaps, or update an existing one, in each
 cluster that will be calling services in remote clusters
 (every cluster in the general case):
 
-For clusters that use `kube-dns`:
-
+{{< tabset cookie-name="platform" >}}
+{{< tab name="KubeDNS" cookie-value="kube-dns" >}}
```
Comment (frankbu) [inline content/en/docs/setup/install/multicluster/gateways/index.md]:
```suggestion
{{< tab name="KubeDNS" cookie-value="kube-dns" >}}

```

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 4]
```diff
@@ -159,9 +159,9 @@ data:
     {"global": ["$(kubectl get svc -n istio-system istiocoredns -o jsonpath={.spec.clusterIP})"]}
 EOF
 {{< /text >}}
+{{< /tab >}}
```
Comment (frankbu) [inline content/en/docs/setup/install/multicluster/gateways/index.md]:
```suggestion

{{< /tab >}}
```

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 5]
```diff
@@ -159,9 +159,9 @@ data:
...
 
-For clusters that use CoreDNS:
-
+{{< tab name="CoreDNS (< 1.4.0)" cookie-value="coredns-prev-1.4.0" >}}
```
Comment (frankbu) [inline content/en/docs/setup/install/multicluster/gateways/index.md]:
```suggestion
{{< tab name="CoreDNS (< 1.4.0)" cookie-value="coredns-prev-1.4.0" >}}

```

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 6]
```diff
@@ -193,6 +193,42 @@ data:
     }
 EOF
 {{< /text >}}
+{{< /tab >}}
```
Comment (frankbu) [inline content/en/docs/setup/install/multicluster/gateways/index.md]:
```suggestion

{{< /tab >}}
```

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 7]
```diff
@@ -193,6 +193,42 @@ data:
...
+
+{{< tab name="CoreDNS (>= 1.4.0)" cookie-value="coredns-after-1.4.0" >}}
+{{< text bash >}}
```
Comment (frankbu) [inline content/en/docs/setup/install/multicluster/gateways/index.md]:
```suggestion

{{< text bash >}}
```

File: content/en/docs/setup/install/multicluster/gateways/index.md [hunk 8]
```diff
@@ -193,6 +193,42 @@ data:
...
+        reload
+        loadbalance
+    }
+    global:53 {
+        errors
+        cache 30
+        forward . $(kubectl get svc -n istio-system istiocoredns -o jsonpath={.spec.clusterIP})
+    }
+EOF
+{{< /text >}}
```
Comment (frankbu) [inline content/en/docs/setup/install/multicluster/gateways/index.md]:
```suggestion
{{< /text >}}

```

**Comment by geeknoid** [comment]
@adammil2000 Aside from the lint errors, are you happy with the PR?

**Comment by johscheuer** [comment]
Fixed the lint issues.

**Comment by adammil2000** [comment]
> @adammil2000 Aside from the lint errors, are you happy with the PR?

LGTM thanks!

**Comment by johscheuer** [comment]
We should add this update to the current docs because it's confusing for most/some users: https://github.com/coredns/coredns/issues/3412

**Pull request merged**